tests/bgp/test_route_aggregation.py

```diff
@@ -0,0 +1,187 @@
+"""
+
+import pytest
+import time
+import logging
+
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+logger = logging.getLogger(__name__)
+
+NEI_IPv4_AGG_ROUTE = "1.1.0.0/16"
+NEI_IPv6_AGG_ROUTE = "1:1:1::/48"
+establish_bgp_session_time = 120
+
+
+def agg_configuration(config, asn, duthost, cli_options, commandv4, commandv6):
+    remove_tag = ""
+    if not config:
+        remove_tag = "no "
+    cmd = 'vtysh{} -c "config" -c "router bgp {}" -c "address-family ipv4 unicast" \
+    -c "{}aggregate-address {} {}" -c "address-family ipv6 unicast" \
+    -c "{}aggregate-address {} {}"'.format(cli_options, asn, remove_tag, NEI_IPv4_AGG_ROUTE, commandv4, remove_tag,
+                                           NEI_IPv6_AGG_ROUTE, commandv6)
+    logger.debug(duthost.shell(cmd, module_ignore_errors=True))
+    cmd = 'vtysh{} -c "clear bgp *"'.format(cli_options)
+    duthost.shell(cmd, module_ignore_errors=True)
+    time.sleep(establish_bgp_session_time)
+
+
+def verify_route_agg(dut, neigh_ip_v4, neigh_ip_v6, num_matches, suppress, as_set=False):
+    output = dut.shell('show ip bgp neighbors {} advertised-routes | grep -c "*> 1.1."'.format(neigh_ip_v4),
+                       module_ignore_errors=True)['stdout']
+    logger.debug(output)
+    assert num_matches == output
+    output = dut.shell('show ipv6 bgp neighbors {} advertised-routes | grep -c "*> 1.1."'.format(neigh_ip_v6),
+                       module_ignore_errors=True)['stdout']
+    logger.debug(output)
+    assert num_matches == output
+
+    output4 = dut.shell('show ip bgp neighbors {} advertised-routes | grep "*> 1.1."'.format(neigh_ip_v4),
+                        module_ignore_errors=True)['stdout']
+    logger.debug(output4)
+    output6 = dut.shell('show ipv6 bgp neighbors {} advertised-routes | grep "*> 1.1."'.format(neigh_ip_v6),
+                        module_ignore_errors=True)['stdout']
+    logger.debug(output6)
+    if suppress:
+        assert "1.1.1.0/24" not in output4
+        assert "1.1.2.0/24" not in output4
+        assert "1:1:1:1::/64" not in output6
+        assert "1:1:1:2::/64" not in output6
+    if as_set:
+        assert "11111" in output4
+        assert "22222" in output4
+        assert "11111" in output6
+        assert "22222" in output6
+
+
+def check_baseline(dut, neigh_ip_v4, neigh_ip_v6, base_v4, base_v6):
+    output = dut.shell('show ip bgp neighbors {} advertised-routes | grep -c "*>"'.format(neigh_ip_v4),
+                       module_ignore_errors=True)['stdout']
+    logger.debug("output: {}".format(output))
+    assert int(base_v4) + 5 == int(output)
+    output = dut.shell('show ipv6 bgp neighbors {} advertised-routes | grep -c "*>"'.format(neigh_ip_v6),
+                       module_ignore_errors=True)['stdout']
+    logger.debug("output: {}".format(output))
+    assert int(base_v6) + 5 == int(output)
+
+
+def test_ebgp_route_aggregation(gather_info):
+    # precheck number of routes
```

Comment:
This test function is a bit too long and difficult to read. Could we encapsulate sections based on their purpose? For example, gathering information could be moved into a separate function that we call from the main routine.

Reply:
I moved a section to its own function to create all the loopbacks to help make it more readable.